crates/rpc-client/src/rpc_client.rs

```diff
@@ -1280,3 +1571,174 @@ impl RPCClient {
         Ok(epoch_number)
     }
 }
+
+#[derive(Debug, PartialEq, Eq, Hash, Clone, Copy)]
+enum CustodianTypeHash {
+    Ckb,
+    Sudt([u8; 32]),
+}
+
+impl CustodianTypeHash {
+    fn is_ckb(&self) -> bool {
+        matches!(self, CustodianTypeHash::Ckb)
+    }
+
+    fn is_sudt(&self) -> bool {
+        matches!(self, CustodianTypeHash::Sudt(_))
+    }
+
+    fn raw(&self) -> [u8; 32] {
```

Comment:
```suggestion
    fn sudt_type_hash(&self) -> Option<[u8; 32]> {
```

We should return Option, let the caller to call result.expect("no sudt type hash")